capplets/desktop-icons/mate-di-capplet.c

```diff
@@ -0,0 +1,380 @@
+		gtk_button_set_label(GTK_BUTTON(capplet->button_computer_toggle), computer_name);
+		g_free(computer_name);
+	}
+	g_variant_unref(user_computer_name);
+
+	/* Get home icon name if set by the user */
+	g_autoptr(GVariant) *user_home_name = g_settings_get_user_value(capplet->desktop_icons_settings, HOME_NAME_KEY);
+	if (user_home_name != NULL)
+	{
+		gchar *home_name = g_settings_get_string(capplet->desktop_icons_settings, HOME_NAME_KEY);
+		gtk_button_set_label(GTK_BUTTON(capplet->button_home_toggle), home_name);
+		g_free(home_name);
+	}
+	g_variant_unref(user_home_name);
+
+	/* Get trash icon name if set by the user */
+	g_autoptr(GVariant) *user_trash_name = g_settings_get_user_value(capplet->desktop_icons_settings, TRASH_NAME_KEY);
+	if (user_trash_name != NULL)
+	{
+		gchar *trash_name = g_settings_get_string(capplet->desktop_icons_settings, TRASH_NAME_KEY);
+		gtk_button_set_label(GTK_BUTTON(capplet->button_trash_toggle), trash_name);
+		g_free(trash_name);
+	}
+	g_variant_unref(user_trash_name);
+
+	/* Get network icon name if set by the user */
+	g_autoptr(GVariant) *user_network_name = g_settings_get_user_value(capplet->desktop_icons_settings, NETWORK_NAME_KEY);
+	if (user_network_name != NULL)
+	{
+		gchar *network_name = g_settings_get_string(capplet->desktop_icons_settings, NETWORK_NAME_KEY);
+		gtk_button_set_label(GTK_BUTTON(capplet->button_network_toggle), network_name);
+		g_free(network_name);
+	}
+	g_variant_unref(user_network_name);
+}
+
+/* Init GUI reading user preferences if set */
+static void
+init_dialog(MateDICapplet* capplet)
+{
+	GtkBuilder* builder;
+	guint builder_result;
+
+	capplet->builder = builder = gtk_builder_new ();
+
+	builder_result = gtk_builder_add_from_resource (builder, "/org/mate/mcc/di/mate-desktop-icons.ui", NULL);
+
+	if (builder_result == 0)
+	{
+		GtkWidget* dialog = gtk_message_dialog_new (NULL, 0, GTK_MESSAGE_ERROR, GTK_BUTTONS_OK, _("Could not load the main interface"));
+		gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog), _("Please make sure that the applet is properly installed"));
+		gtk_dialog_set_default_response (GTK_DIALOG (dialog), GTK_RESPONSE_OK);
+
+		gtk_dialog_run(GTK_DIALOG(dialog));
+
+		gtk_widget_destroy(dialog);
+		exit(EXIT_FAILURE);
+	}
+
+	capplet->window = get_widget("desktop_icons_dialog");
+
+	/* Settings */
+	capplet->show_icons_settings = g_settings_new (SHOW_ICONS_SCHEMA);
+	capplet->desktop_icons_settings = g_settings_new (DESKTOP_ICONS_SCHEMA);
+
+	capplet->button_show_icons_toggle = get_widget("button_show_icons_toggle");
+	capplet->desktop_icons_caption = get_widget("desktop_icons_caption");
+
+	/* CheckButton to allow icon showing or not */
+	gtk_toggle_button_set_active(capplet->button_show_icons_toggle, g_settings_get_boolean(capplet->show_icons_settings, SHOW_ICONS_KEY));
+
+	capplet->desktop_icons_caption = get_widget("desktop_icons_caption");
```

Comment:
**cppcheck warning**
Id: redundantAssignment (see line 294)
Variable 'capplet->desktop_icons_caption' is reassigned a value before the old one has been used.